docs/contributing/test-data-models-overview.md

```diff
@@ -0,0 +1,250 @@
+# Test data models overview
+
+A test data model is set of helper objects commercetools API consumers can use during writing their own tests in order to mock the data returned by our services.
+
+These helpers are flexible enough to represent different types of responses from different APIs and with different values.
+
+## Public API
+
+Each test data model has the same API which consists on:
+
+- A Rest builder
+- A Graphql builder
+- Typescript types
+
+A `Builder` is the object you would be interacting with as a consumer and it has only two functions:
+
+- `random`: exposes a fluent API which allows to build a customized version of the data model
+- `presets`: predefined versions of the data model
+
+This is how it looks like the entry point of a data model package:
+
+```ts
+import { RestModelBuilder, GraphqlModelBuilder } from './builders';
+import * as GeometryPresets from './presets';
+
+export * from './types';
+
+export const GeometryRest = {
+  random: RestModelBuilder,
+  presets: GeometryPresets.restPresets,
+};
+
+export const GeometryGraphql = {
+  random: GraphqlModelBuilder,
+  presets: GeometryPresets.graphqlPresets,
+};
+```
+
+### Examples
+
+#### Building default random data models
```

Comment:
Nit, would have been easier for me to follow this way

```suggestion
#### Building fully random data models
```

Reply:
My main issue here is that "fully" would imply to me that the data model will have all its values populated, which is not what actually happens.
So far, I've seen we're populating the the required fields in the generators and, for some models, some other fields, but we never return a fully populated object from a simple `Model.random().build()`.

Reply:
Got it. Makes sense. Thanks for explaining.